smac/intensification/intensification.py

```diff
@@ -515,16 +920,96 @@ def _get_instances_to_run(self,
 
         return to_run, inc_sum_cost
 
-    def get_next_challenger(self,
-                            challengers: typing.Optional[typing.List[Configuration]],
-                            chooser: typing.Optional[EPMChooser],
-                            run_history: typing.Optional[RunHistory] = None,
-                            repeat_configs: bool = True) -> \
+    def query_challenger(self,
```

Comment:
I feel that this function currently does more than one thing which I find really confusing. Suggestion: at the beginning of a new iteration we call the function to create new challengers from the challenger list or the chooser, this would reduce the amount of different things this function does and hopefully make it a bit easier to comprehend.

Reply:
I tried simplifying the function so that now it is needed when you desired a new challenger, not in all possible situations as it was before. Please let me know if you consider it should be simplified more.

Conceptually I think this function was meant to be used to get the next challenger for racing (it is taken from the master branch), but it could be called at any time by anyone, so ti was a bit flexible, but un-readable.

I removed this flexibility and updated the doc accordingly.